src/test/java/io/lettuce/core/failover/metrics/LockFreeSlidingTimeWindowMetricsUnitTests.java

```diff
@@ -193,21 +193,45 @@ void shouldBeThreadSafe() throws InterruptedException {
         assertThat(snapshot.getTotalCount()).isEqualTo(threadCount * eventsPerThread);
     }
 
+    /**
+     * Current implementation does not support single bucket window. This test ensures proper tracking with minimum window size
+     * (2 buckets). NPE will be thrown if used with windowSize=1, after second full window rotation.
+     */
     @Test
-    @DisplayName("should track events on window with single bucket")
-    void shouldTrackEventsOnWindowWithSingleBucket() {
-        int windowSize = BUCKET_SIZE * 1;
+    @DisplayName("should track events with minimum window size (2*buckets)")
+    void shouldTrackEventsOnMinimumWindowSize() {
+        int windowSize = BUCKET_SIZE * 2;
         LockFreeSlidingTimeWindowMetrics metrics = new LockFreeSlidingTimeWindowMetrics(windowSize, clock);
-        metrics.recordSuccess();
-        metrics.recordFailure();
-        MetricsSnapshot snapshot = metrics.getSnapshot();
-        assertThat(snapshot.getSuccessCount()).isEqualTo(1);
-        assertThat(snapshot.getFailureCount()).isEqualTo(1);
 
-        // advance one bucket
-        clock.advance(BUCKET_SIZE_DURATION);
+        clock.advance(Duration.ofMillis(100));
+        // metrics.recordSuccess();
+        assertThat(metrics.getSnapshot().getSuccessCount()).isEqualTo(0);
+
+        clock.advance(Duration.ofSeconds(1));
+        clock.advance(Duration.ofMillis(1));
         assertThat(metrics.getSnapshot().getSuccessCount()).isEqualTo(0);
-        assertThat(metrics.getSnapshot().getFailureCount()).isEqualTo(0);
+
+        metrics.recordSuccess();
+        assertThat(metrics.getSnapshot().getSuccessCount()).isEqualTo(1);
+
+        // Create metrics with window size 1 - creates single node with next=null
```

Comment:
```suggestion
        // Create metrics with window size **2** - creates single node with next=null
```

and others that mention "1 sec".